src/modules/experts/components/ExpertsView.tsx

```diff
@@ -35,44 +50,104 @@ const ExpertsView: React.FC = () => {
     (state: RootStateType) => state.properties?.directions,
   );
 
-  const expertsPropertiesLoaded = !!regions.length && !!directions.length;
+  const fetchMaterials = () => {
+    const pageQuery = Number(query.get(PAGE_QUERY));
+    const regionsQuery = query.get(REGIONS_QUERY);
+    const DirectionsQuery = query.get(DIRECTIONS_QUERY);
+
+    const isPage = pageQuery ? pageQuery - 1 : 0;
+    dispatch(setExpertsPage(isPage));
+
+    const regionsType = {};
+    regionsQuery?.split(',').forEach((el) => {
+      regionsType[el] = true;
+    });
+    dispatch(setExpertsRegionsFilter({ value: regionsType }));
+
+    const directionsType = {};
+    DirectionsQuery?.split(',').forEach((el) => {
+      directionsType[el] = true;
+    });
+    dispatch(setExpertsDirectionsFilter({ value: directionsType }));
+
+    dispatch(fetchExperts());
+  };
+
+  const setFilters = (
+    checked: ICheckBoxFormState,
+    filterType: FilterTypeEnum,
+  ) => {
+    const checkedIds = Object.keys(checked).filter((key) => checked[key]);
+    const queryType = filterType.toLowerCase();
 
-  const handlePageChange = (_, page: number) => {
-    if (page === 1) {
-      dispatch(setExpertsPage(0));
+    query.set(queryType, checkedIds.join(','));
+    if (!checkedIds.length) {
+      query.delete(queryType);
     } else {
-      dispatch(setExpertsPage(page - 1));
+      query.set(queryType, checkedIds.join(','));
     }
```

Comment:
else statement seems obsolete, it's already on line : 83

Reply:
Fixed and improved